### PR TITLE
Remove schema and AST from the IR

### PR DESCRIPTION
**Pull request by martinbonnin**
Prep-work for schemaBased builders, cleanup the IR by removing the schema and AST types from there. This allows to factor some of the logic between Java/Kotlin codegen and more generally, it's better separation of concerns

### CODE REVIEW

File: apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/DefaultApolloStore.kt [hunk 1]
```diff
@@ -201,6 +201,7 @@ internal class DefaultApolloStore(
           cacheKeyGenerator = objectIdGenerator
       )
 
+      records.flatMap { it.value.fieldKeys() }
```
Comment (BoD) [inline apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/DefaultApolloStore.kt:204]: Not intentional?

**Review by BoD — APPROVED** [review]
⚙️

File: apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/DefaultApolloStore.kt (hunk 1 above)
Comment (martinbonnin) [reply apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/DefaultApolloStore.kt:204]: uh oh good catch 👍

**Comment by martinbonnin** [comment]
As it's a relatively big PR, I'm keeping this one open until we release 3.5.0

**Pull request merged**